src/Compatibility/Microsoft.DotNet.PackageValidation/Microsoft.DotNet.PackageValidation.csproj

```diff
@@ -0,0 +1,92 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>$(SdkTargetFramework);net472</TargetFrameworks>
+    <IsPackable>true</IsPackable>
+    <IsShipping>true</IsShipping>
+    <StrongNameKeyId>Open</StrongNameKeyId>
+    <PackageType>MSBuildSdk</PackageType>
+    <VersionPrefix>0.1.0</VersionPrefix>
+    <IncludeBuildOutput>false</IncludeBuildOutput>
+    <NoWarn>$(NoWarn);NU5128</NoWarn>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\..\Tasks\Common\**\*.cs" LinkBase="Common" />
+    <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildFrameworkVersion)" ExcludeAssets="Runtime" />
+    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildUtilitiesCoreVersion)" ExcludeAssets="Runtime" />
+    <PackageReference Include="NuGet.Frameworks" Version="$(NuGetBuildTasksPackageVersion)" />
+    <PackageReference Include="NuGet.Packaging" Version="$(NuGetBuildTasksPackageVersion)" />
+    <PackageReference Include="NuGet.Protocol" Version="$(NuGetBuildTasksPackageVersion)" />
+    <ProjectReference Include="..\Microsoft.DotNet.ApiCompatibility\Microsoft.DotNet.ApiCompatibility.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="..\..\Tasks\Common\Resources\Strings.resx" LinkBase="Resources" GenerateSource="True" Namespace="Microsoft.NET.Build.Tasks" />
+    <EmbeddedResource Update="Resources.resx" GenerateSource="true" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <InternalsVisibleTo Include="Microsoft.DotNet.PackageValidation.Tests" />
+  </ItemGroup>
+
+  <!--
+    Default to including all *.props and *.targets files
+    from the project directory into the NuGet package root
+  -->
+  <ItemGroup>
+    <None Include="**/*.props;**/*.targets" Pack="true">
+      <PackagePath>%(RecursiveDir)%(Filename)%(Extension)</PackagePath>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Include="$(RepoRoot)LICENSE.txt" PackagePath="LICENSE.txt" Pack="true"/>
+  </ItemGroup>
+
+  <PropertyGroup>
+    <TargetsForTfmSpecificContentInPackage>$(TargetsForTfmSpecificContentInPackage);_AddBuildOutputToPackageCore;_AddBuildOutputToPackageDesktop</TargetsForTfmSpecificContentInPackage>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!--
+      Update all PackageReference and ProjectReference Items to have
+      PrivateAssets="All" and default Publish to true.
+      This removes the dependency nodes from the generated nuspec and
+      forces the publish output to contain the dlls.
+     -->
+    <PackageReference Update="@(PackageReference)">
```

Comment:
How are we going to load Roslyn and NuGet as part of the task? I think we should load them as part of the constructor of the task and maybe point the task to the SDK (MSBuild) path and then resolve from them. 

I did something like that on my prototype task to run APICompat: https://github.com/safern/apicompat/blob/d5f14eb6695c299f2f4a379cdc47c5cb1c781806/src/Microsoft.DotNet.ApiCompatibility.Tasks/ApiCompatibility.cs#L17

However we need to be careful to make sure to always run on the latest version. Maybe we can carry these as part of the task as a "fallback" folder that if we didn't find a good enough version, then fallback and load ours?

cc: @ericstj

Reply:
I'd be curious if there are any other design time components that run outside the compiler process but need to load roslyn.  IMHO the SDK version should always be good enough, we might want to use the version referenced by a project if it happens to reference a newer version.

> Maybe we can carry these as part of the task as a "fallback" folder that if we didn't find a good enough version, then fallback and load ours?

What sort of version wouldn't be good enough?  Do we need new APIs or are you just worried about assembly version?

Reply:
> What sort of version wouldn't be good enough? Do we need new APIs or are you just worried about assembly version?

My concern was more around if we depended on metadata that the current roslyn API wouldn't understand, i.e if a new feature is added and for that we need a new API but the project that is using our tools is running on an older SDK and hence that roslyn version will not have that API available.